app/domain/authentication/jwt/decoded_credentials.rb

```diff
@@ -6,21 +6,44 @@ class DecodedCredentials
       JWT_REQUEST_BODY_FIELD_NAME = "jwt".freeze
 
       def initialize(credentials)
+        @logger = Rails.logger
         @decoded_credentials = Hash[URI.decode_www_form(credentials)]
-        validate
+        validate_jwt_request_field_is_present
+        validate_body_contains_jwt
       end
 
       def jwt
-        @decoded_credentials[JWT_REQUEST_BODY_FIELD_NAME]
+        @jwt_token
       end
 
       private
 
-      def validate
+      def validate_jwt_request_field_is_present
         if @decoded_credentials.fetch(JWT_REQUEST_BODY_FIELD_NAME, "") == ""
           raise Errors::Authentication::RequestBody::MissingRequestParam, JWT_REQUEST_BODY_FIELD_NAME
         end
+        @jwt_token = @decoded_credentials[JWT_REQUEST_BODY_FIELD_NAME].strip
       end
+
+      def validate_body_contains_jwt
+        raise Errors::Authentication::Jwt::RequestBodyIsNotJWTToken unless is_jwt?
+      end
+
+      def is_jwt?
+        @jwt_token =~ jwt_regex
+      end
+
+      def jwt_regex
+        # https://datatracker.ietf.org/doc/html/rfc4648#section-5
+        # jwt token is 3 blocks of base64url encoded strings separated by period '.'
+        # base64url encoding differs from regular base64 encoding as follows
+        # - padding is skipped so the pad character '=' doesn't have to be percent encoded
+        # - the 62nd and 63rd regular base64 encoding characters ('+' and '/') are replace with ('-' and '_')
+        #   The changes make the encoding alphabet file and URL safe.
+        # tokens without a signature part are denied
+        /\A[-A-Za-z0-9_=]+\.[-A-Za-z0-9_=]+\.[-A-Za-z0-9_=]+\z/
```

Comment:
I think it is better to define it as const instead of as a method implementation.

Reply:
Also, consider taking out the 3 similar parts that match the base64url (that is, the `[-A-Za-z0-9_=]+`) into a separate regex const so at the end you will have:
```suggestion
        /\A#{BASE_64_URL_REGEX}\.#{BASE_64_URL_REGEX}\.#{BASE_64_URL_REGEX}\z/
```
which is more readable and maintainable IMO.